Lib/test/_test_multiprocessing.py

```diff
@@ -4472,17 +4473,34 @@ def check_semaphore_tracker_death(self, signum, should_die):
         # bpo-31310: if the semaphore tracker process has died, it should
         # be restarted implicitly.
         from multiprocessing.semaphore_tracker import _semaphore_tracker
-        _semaphore_tracker.ensure_running()
         pid = _semaphore_tracker._pid
+        if pid:
+            os.kill(pid, signal.SIGKILL)
+            time.sleep(0.5)  # give it time to die
+        old_stderr = sys.stderr
+        r, w = os.pipe()
+        try:
```

Comment:
Notice that we cannot use  `test.support.captured_stderr()` as it does not support `fileno()`.